src/main/java/dev/openfeature/sdk/OpenFeatureAPI.java

```diff
@@ -1,43 +1,47 @@
 package dev.openfeature.sdk;
 
-import lombok.Getter;
-import lombok.Setter;
-
-import javax.annotation.Nullable;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
 
+import javax.annotation.Nullable;
+
+import dev.openfeature.sdk.internal.AutoCloseableLock;
+import dev.openfeature.sdk.internal.AutoCloseableReentrantReadWriteLock;
+import lombok.Getter;
+import lombok.Setter;
+
 /**
  * A global singleton which holds base configuration for the OpenFeature library.
  * Configuration here will be shared across all {@link Client}s.
  */
 public class OpenFeatureAPI {
     private static OpenFeatureAPI api;
+
+    // package-private multi-read/single-write lock
+    static AutoCloseableReentrantReadWriteLock rwLock = new AutoCloseableReentrantReadWriteLock();
     @Getter
-    @Setter
     private FeatureProvider provider;
     @Getter
     @Setter
     private EvaluationContext evaluationContext;
     @Getter
     private List<Hook> apiHooks;
 
-    public OpenFeatureAPI() {
-        this.apiHooks = new ArrayList<>();
+    private static class SingletonHolder {
+        private static final OpenFeatureAPI INSTANCE = new OpenFeatureAPI();
     }
 
     /**
      * Provisions the {@link OpenFeatureAPI} singleton (if needed) and returns it.
      * @return The singleton instance.
      */
     public static OpenFeatureAPI getInstance() {
-        synchronized (OpenFeatureAPI.class) {
-            if (api == null) {
-                api = new OpenFeatureAPI();
-            }
-        }
-        return api;
+        return SingletonHolder.INSTANCE;
+    }
```

Comment:
This is a more performant singleton. The `synchronized` block in the previous implementation involved inherent locking on _each call_. The "double check" singleton implementation can help here, but it also has it's downfalls.

I've used the `Bill Pugh" approach here.

This takes advantage of basic Java spec requirements that guarantee classes are loaded only once. There's no synchronization concerns here and no performance penalty.